src/Functions/FunctionCustomWeekToSomething.h

```diff
@@ -143,15 +137,15 @@ class FunctionCustomWeekToSomething : public IFunction
 
         if (checkAndGetDataType<DataTypeDate>(&type))
         {
-            return Transform::FactorTransform::execute(UInt16(left.get<UInt64>()), DEFAULT_WEEK_MODE, date_lut)
-                    == Transform::FactorTransform::execute(UInt16(right.get<UInt64>()), DEFAULT_WEEK_MODE, date_lut)
+            return Transform::execute(UInt16(left.get<UInt64>()), DEFAULT_WEEK_MODE, date_lut)
+                    == Transform::execute(UInt16(right.get<UInt64>()), DEFAULT_WEEK_MODE, date_lut)
```

Comment:
This looks wrong, because, it will erroneously return true for ranges, that start at the same week, but last more than year and will return false for ranges, that last less than one year, but start at different weeks. I think, it's better to use `ToStartOfYearImpl` as `FactorTransform` for `toWeek` like it is done, e.g. for function `toMonth` in `DateTimeTransforms.h`.

Reply:
Yeah. I didn't understand the code style. It should be trivial :) Updated